DockService.App/Messaging/DockEventHandler.cs

```diff
@@ -0,0 +1,70 @@
+﻿using DockService.Core.Messaging;
+using DockService.Core.Models;
+using DockService.Core.Services;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Utf8Json;
+
+namespace DockService.App.Messaging
+{
+    public class DockEventHandler : IEventHandlerCallback
+	{
+		private readonly IDockService _dockService;
+        public DockEventHandler(IDockService dockService)
+        {
+            _dockService = dockService;
+        }
+
+        #region --- main stuff---
+        public async Task<bool> HandleEventAsync(EventTypes eventType, string message)
+        {
+            switch (eventType)
+            {
+                case EventTypes.ShipNearingHarbor:
+                    {
+                        return await HandleShipNearingHarbor(message);
+                    }
+                case EventTypes.ShipUndock:
+                    {
+                        return await HandleShipUndock(message);
```

Comment:
Why the method name is called: ShipUndocked instead of: ShipUndockRequested ?

Reply:
because they are events. not requests

Reply:
Tho we should rename it to ShipReadyForUndocking i just thought of that. That does require major refactoring acrouss multiple services